File: examples/v2/stack/python/__main__.py
import pulumi
import pulumi_pulumiservice.v2 as ps_v2

config = pulumi.Config()
service_org = config.get("serviceOrg") or "service-provider-test-org"
project_name = config.get("projectName") or "pulumi-service-stack-example"
stack_name = config.get("stackName") or "dev"
stack_purpose = config.get("stackPurpose") or "demo"

example_stack = ps_v2.Stack(
    "exampleStack",
    org_name=service_org,
    project_name=project_name,
    stack_name=stack_name,
    tags={
        "owner": "pulumicloud-v2-example",
        "purpose": stack_purpose,
    },
)

pulumi.export("stackName", example_stack.stack_name)

File: examples/v2/stack/go/main.go
package main

import (
	stacks "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v2/stacks"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		serviceOrg := cfg.Get("serviceOrg")
		if serviceOrg == "" {
			serviceOrg = "service-provider-test-org"
		}
		projectName := cfg.Get("projectName")
		if projectName == "" {
			projectName = "pulumi-service-stack-example"
		}
		stackName := cfg.Get("stackName")
		if stackName == "" {
			stackName = "dev"
		}
		stackPurpose := cfg.Get("stackPurpose")
		if stackPurpose == "" {
			stackPurpose = "demo"
		}

		exampleStack, err := stacks.NewStack(ctx, "exampleStack", &stacks.StackArgs{
			OrgName:     pulumi.String(serviceOrg),
			ProjectName: pulumi.String(projectName),
			StackName:   pulumi.String(stackName),
			Tags: pulumi.Map{
				"owner":   pulumi.String("pulumicloud-v2-example"),
				"purpose": pulumi.String(stackPurpose),
			},
		})
		if err != nil {
			return err
		}

		ctx.Export("stackName", exampleStack.StackName)
		return nil
	})
}
